# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1292379476 28800
#      Tue Dec 14 18:17:56 2010 -0800
# Node ID 4cab008d3be12ae8386063023d5435b49a9190e3
# Parent  a0dfd7e8e3e2505cafc2b26fff173059ed9ed757
Use high contrast color black for arrows in quiver3 plots.  Bug #31802.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2010-12-14  John W. Eaton  <jwe@octave.org>
+
+	* plot/private/__quiver__.m: Use high contrast color black for arrows
+	in quiver3 plots.  Bug #31802.
+
 2010-12-14  Rik  <octave@nomad.inbox5.com>
 
 	* plot/compass.m, plot/errorbar.m, plot/legend.m, plot/rose.m, 
 	plot/scatter.m, plot/scatter3.m, plot/stairs.m: Assign data used in 
 	demo plots for reproducibility between runs.
 
 2010-12-13  Rik  <octave@nomad.inbox5.com>
 
diff --git a/scripts/plot/private/__quiver__.m b/scripts/plot/private/__quiver__.m
--- a/scripts/plot/private/__quiver__.m
+++ b/scripts/plot/private/__quiver__.m
@@ -187,17 +187,17 @@ function hg = __quiver__ (varargin)
                    "linestyle", linespec.linestyle, 
                     "color", linespec.color, "parent", hg);
       endif
     else
       if (is3d)
         h1 = plot3 ([x.'; xend.'; NaN(1, length (x))](:),
                     [y.'; yend.'; NaN(1, length (y))](:),
                     [z.'; zend.'; NaN(1, length (z))](:),
-                    "parent", hg);
+                    "color", "black", "parent", hg);
       else
         h1 = plot ([x.'; xend.'; NaN(1, length (x))](:),
                    [y.'; yend.'; NaN(1, length (y))](:),
                    "parent", hg);
       endif
     endif
 
     xtmp = x + uu(:) .* (1 - arrowsize);
@@ -236,17 +236,17 @@ function hg = __quiver__ (varargin)
                      "linestyle", linespec.linestyle,
                       "color", linespec.color, "parent", hg);
         endif
       endif
     elseif (is3d)
       h2 = plot3 ([xarrw1.'; xend.'; xarrw2.'; NaN(1, length (x))](:),
                   [yarrw1.'; yend.'; yarrw2.'; NaN(1, length (y))](:),
                   [zarrw1.'; zend.'; zarrw2.'; NaN(1, length (z))](:),
-                  "parent", hg);
+                  "color", "black", "parent", hg);
     else
       h2 = plot ([xarrw1.'; xend.'; xarrw2.'; NaN(1, length (x))](:),
                  [yarrw1.'; yend.'; yarrw2.'; NaN(1, length (y))](:),
                  "parent", hg);
     endif
 
     if (! have_line_spec
         || (isfield (linespec, "marker")
